src/NuGet.Clients/NuGet.PackageManagement.UI/Models/Package/PackageModel.cs

```diff
@@ -89,10 +94,21 @@ internal PackageModel(PackageIdentity identity,
 
         public Uri? ReadmeUri => _embeddedResources.ReadmeUri;
 
+        public IReadOnlyList<PackageVulnerabilityMetadataContextInfo>? Vulnerabilities => _vulnerableCapability.Vulnerabilities;
+
+        public bool IsVulnerable => _vulnerableCapability.IsVulnerable;
+
+        public PackageVulnerabilitySeverity VulnerabilityMaxSeverity => _vulnerableCapability.VulnerabilityMaxSeverity;
+
         public ValueTask<Stream?> GetIconAsync(CancellationToken cancellationToken) => _embeddedResources.GetIconAsync(cancellationToken);
 
         public ValueTask<Stream?> GetLicenseAsync(CancellationToken cancellationToken) => _embeddedResources.GetLicenseAsync(cancellationToken);
 
         public ValueTask<Stream?> GetReadmeAsync(CancellationToken cancellationToken) => _embeddedResources.GetReadmeAsync(cancellationToken);
+
+        public virtual async Task PopulateDataAsync(CancellationToken cancellationToken)
```

Comment:
This should probably be abstract so each package type can call the appropriate updates. Unless there's a capability that's in the base.